src/api/grainConfig.js

```diff
@@ -39,6 +49,17 @@ export function toDistributionPolicy(x: GrainConfig): DistributionPolicy {
       policyType: "IMMEDIATE",
     });
   }
+  if (x.recentPerWeek > 0) {
+    const discount = NullUtil.orElse(x.discount, undefined);
+    if (discount === undefined) {
+      throw new Error(`no discount specified for recent policy`);
+    }
+    allocationPolicies.push({
+      budget: G.fromInteger(x.immediatePerWeek),
```

Comment:
there's a subtle and serious bug on this line. maybe consider adding tests to catch this issue. the only reason i don't insist on it is that i agree that we should refactor this config entirely.

Reply:
Thanks for pointing this out.  Added another test to cover this and check values directly before/after.  In general this module is lacking on tests, so I'll try to populate it with more as I go along.